File: backend/src/main/java/com/smartcampus/backend/dto/parking/ParkingStatusDTO.java
package com.smartcampus.backend.dto.parking;

import lombok.*;
import java.time.OffsetDateTime;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class ParkingStatusDTO {
    private Long id;
    private String ident;
    private String nom;
    private String adresse;
    private String exploit;
    private String infor;
    private String taType;
    private String type;
    private Double latitude;
    private Double longitude;

    // Capacités
    private Integer npTotal;
    private Integer npGlobal;
    private Integer npPmr;
    private Integer npVle;
    private Integer npVeltot;
    private Integer npVelec;
    private Integer np2rmot;
    private Integer npCovoit;

    // Tarifs horaires (th)
    private Double thQuar;
    private Double thDemi;
    private Double thHeur;
    private Double th2;
    private Double th3;
    private Double th4;
    private Double th10;
    private Double th24;
    private Double thNuit;

    // Tarifs abonnements (ta)
    private Double taTitul;
    private Double taNtitul;
    private Double taResmoi;
    private Double taNres7j;
    private Double taMoimot;
    private Double taMoivel;
    private String taHandi;

    // Infra
    private String anServ;
    private String secteur;
    private String propr;
    private String typgest;
    private Integer nbNiv;
    private Double gabariStd;
    private Double gabariMax;

    // Données dynamiques (Temps Réel)
    private String etat;
    private Integer libre;
    private Integer totalTempsReel;
    private Boolean connecte;
    private OffsetDateTime mdate;
    private boolean dataFraiche; 

    private String url;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/parking/ParkingStatusDto.kt
package com.ObservatoireCampus.mobile.model.parking

data class ParkingStatusDto(
    val id: Long,
    val ident: String,
    val nom: String,
    val adresse: String?,
    val exploit: String?,
    val infor: String?,
    val taType: String?,
    val type: String?,          // SURFACE / SILO / ENTERRE / MIXTE
    val latitude: Double?,
    val longitude: Double?,

    // Capacités
    val npTotal: Int?,
    val npGlobal: Int?,
    val npPmr: Int?,
    val npVle: Int?,
    val npVeltot: Int?,
    val npVelec: Int?,
    val np2rmot: Int?,
    val npCovoit: Int?,

    // Tarifs horaires (th)
    val thQuar: Double?,
    val thDemi: Double?,
    val thHeur: Double?,
    val th2: Double?,
    val th3: Double?,
    val th4: Double?,
    val th10: Double?,
    val th24: Double?,
    val thNuit: Double?,

    // Tarifs abonnements (ta)
    val taTitul: Double?,
    val taNtitul: Double?,
    val taResmoi: Double?,
    val taNres7j: Double?,
    val taMoimot: Double?,
    val taMoivel: Double?,
    val taHandi: String?,

    // Caractéristiques Infra
    val anServ: String?,
    val secteur: String?,
    val propr: String?,
    val typgest: String?,
    val nbNiv: Int?,
    val gabariStd: Double?,
    val gabariMax: Double?,

    // Données dynamiques (Temps Réel)
    val etat: String?,          // OUVERT / FERME / COMPLET
    val libre: Int?,
    val totalTempsReel: Int?,
    val connecte: Boolean?,
    val mdate: String?,
    val dataFraiche: Boolean,

    val url: String?
)